glue/core/visual.py

```diff
@@ -109,13 +118,21 @@ def preferred_cmap(self):
     def preferred_cmap(self, value):
         if isinstance(value, str):
             try:
-                for i, element in enumerate(colormaps.members):
+                self._preferred_cmap = get_cmap(value)
+            except ValueError:
+                # This checks for the formal name of the colormap.
+                # e.g., 'viridis' is 'Viridis'
+                for element in colormaps.members:
                     if element[0] == value:
                         self._preferred_cmap = element[1]
-            except TypeError:
-                self._preferred_cmap = get_cmap(value)
-        else:
+                        break
+                else:
+                    # If the string name fails to be validated
+                    raise ValueError(f"{value} is not a valid colormap name.")
+        elif isinstance(value, Colormap) or value is None:
             self._preferred_cmap = value
+        else:
+            raise ValueError("`preferred_cmap` must be a string or an instance of a matplotlib.colors.Colormap")
```

Comment:
```suggestion
            raise TypeError("`preferred_cmap` must be a string or an instance of a matplotlib.colors.Colormap")
```
this was lost somewhere in the pushes; technical, but quite a clear case of https://docs.python.org/3/library/exceptions.html#TypeError.

Reply:
Ah I must have not noticed that and forgot to change it.